waku/node/waku_node.nim

```diff
@@ -223,62 +236,104 @@ proc registerRelayDefaultHandler(node: WakuNode, topic: PubsubTopic) =
     await filterHandler(topic, msg)
     await archiveHandler(topic, msg)
 
-  node.wakuRelay.subscribe(topic, defaultHandler)
+  discard node.wakuRelay.subscribe(topic, defaultHandler)
 
-
-proc subscribe*(node: WakuNode, topic: PubsubTopic) =
+proc subscribe*(node: WakuNode, subscription: SubscriptionEvent) =
+  ## Subscribe to a pubsub topic without a handler.
+  ## Adds default handler to push messages to FILTER & STORE
+  
   if node.wakuRelay.isNil():
     error "Invalid API call to `subscribe`. WakuRelay not mounted."
     return
+  
+  let (pubsubTopic, _) = getSubscriptionShard(subscription)
 
-  debug "subscribe", pubsubTopic= topic
+  if node.wakuRelay.isSubscribed(pubsubTopic):
+    error "Invalid API call to `subscribe`. Was already subscribed"
+    return
 
-  node.topicSubscriptionQueue.emit(SubscriptionEvent(kind: PubsubSub, pubsubSub: topic))
-  node.registerRelayDefaultHandler(topic)
+  debug "subscribe", pubsubTopic=pubsubtopic
 
-proc subscribe*(node: WakuNode, topic: PubsubTopic, handler: WakuRelayHandler) =
-  ## Subscribes to a PubSub topic. Triggers handler when receiving messages on
-  ## this topic. TopicHandler is a method that takes a topic and some data.
+  node.topicSubscriptionQueue.emit((kind: PubsubSub, topic: pubsubtopic))
+  node.registerRelayDefaultHandler(pubsubtopic)
+
+proc subscribe*(node: WakuNode, subscription: SubscriptionEvent, handler: WakuRelayHandler) =
```

Comment:
I know people have different opinions about this overloading approach, but why not make this:

```suggestion
proc subscribe*(node: WakuNode, subscription: SubscriptionEvent, handler: Option[WakuRelayHandler]) =
```

The `subscribe*` in lines 241 to 258 can then simply redirect to this proc with:

```
proc subscribe*(node: WakuNode, subscription: SubscriptionEvent) =
  node.subscribe(subscription, none(WakuRelayHandler)
```

and you wouldn't need to duplicate the `subscribe` logic.

Reply:
True! I'll do that